src/roslibpy/ros.py

```diff
@@ -180,6 +180,57 @@ def _send_internal(proto):
 
         self.factory.on_ready(_send_internal)
 
+    def blocking_call_from_thread(self, callback, timeout):
+        """Call the given function from a thread, and wait for the result synchronously
+        for as long as the timeout will allow.
+
+        Args:
+            callback: Callable function to be invoked from the thread.
+            timeout (:obj: int): Number of seconds to wait for the response before
+                raising an exception.
+
+        Returns:
+            The results from the callback, or a timeout exception.
+        """
+        return self.factory.manager.blocking_call_from_thread(callback, timeout)
+
+    def get_service_request_callback(self, message):
+        """Get the callback which, when called, sends the service request.
+
+        Args:
+            message (:class:`.Message`): ROS Bridge Message containing the request.
+
+        Returns:
+            A callable which makes the service request.
+        """
+        def get_call_results(result_placeholder):
+
+            inner_callback = self.factory.manager.get_inner_callback(result_placeholder)
+
+            inner_errback = self.factory.manager.get_inner_errback(result_placeholder)
+
+            self.send_service_request(message, inner_callback, inner_errback)
+
+            return result_placeholder
+
+        return get_call_results
+
+    def make_blocking_service_request(self, message, timeout):
```

Comment:
Would it make more sense to call this `send_blocking_service_request` to be closer to the existing `send_service_request` method?

Reply:
I had considered that, but had decided that because we are also returning a response that a distinction should be made between the two verbs.  I am happy to change it though.

Reply:
Ah, ok! 
Perhaps we have misnamed `send_service_request` to begin with. It was named like that to match the `send_message`/`send_close` methods found in other parts of the library, but perhaps we should rename to `call_service`?

Reply:
Makes sense to me.  Can do.